src/pages/Models/Details/ModelDetails.js

```diff
@@ -32,59 +32,69 @@ import "./_model-details.scss";
   @param {String} appName The name of the application to filter the data by.
 */
 const filterModelStatusData = (modelStatusData, appName) => {
-  if (!modelStatusData || appName === "") {
-    return modelStatusData;
-  }
-  const filteredData = cloneDeep(modelStatusData);
-  const application = filteredData.applications[appName];
-  // remove the units from the application objects that are not
-  // the filter-by app.
-  const subordinateTo = application.subordinateTo || [];
-  Object.keys(filteredData.applications).forEach((key) => {
-    if (key !== appName && !subordinateTo.includes(key)) {
-      filteredData.applications[key].units = {};
+  if (modelStatusData) {
+    const filteredData = cloneDeep(modelStatusData);
+    Object.keys(filteredData.applications).forEach((key) => {
```

Comment:
I feel like this logic belongs in the selector as anywhere that requests this data may find this number useful. If you make the change there then you don't have to modify this file at all.

Reply:
Won't it be better to wait until something else needs it before escalating to a selector?

Reply:
I agree that in this type of situation It's usually a good idea to wait for another use case but then you run into the problem when someone needs the unit count, will they know that this already exists and need to go update it?

In this specific situation however I know that we'll need the unit count with the new 'details' pages that are coming so we might as well head that off right now.

Reply:
We've discussed this and decided to leave it here for now.